src/main/java/io/camunda/testing/extensions/ZeebeAssertionsExtension.java

```diff
@@ -11,50 +11,66 @@
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.platform.commons.util.ReflectionUtils;
 
-// TODO rewrite this mess
 public class ZeebeAssertionsExtension implements BeforeEachCallback, AfterEachCallback {
 
   private static final String KEY_ZEEBE_CLIENT = "ZEEBE_CLIENT";
 
   @Override
   public void beforeEach(final ExtensionContext extensionContext) throws Exception {
+    final Object testInstance = extensionContext.getRequiredTestInstance();
```

Comment:
🙃  For understandability, I'd move this line into both of the store methods, duplicated just like the `getRequiredTestClass()` calls. That way the beforeEach becomes more descriptive, first do this, then do that.